test/command_checker/add_reference_test.rb

```diff
@@ -138,6 +138,30 @@ def test_add_reference_foreign_key_no_validate
       assert_safe AddReferenceForeignKeyNoValidate
     end
 
+    def test_add_reference_foreign_key_no_validate_is_not_idempotent
+      assert_empty @connection.foreign_keys(:projects)
+
+      migrate AddReferenceForeignKeyNoValidate
+      assert_raises_with_message(StandardError, /column "user_id" of relation "projects" already exists/) do
```

Comment:
I think this is expected behaviour, so I added this as a passing test (though I guess it would be nice if it was consistent with `add_reference_concurrently`).